src/colorizer/loaders/UrlArrayLoader.ts

```diff
@@ -71,8 +73,11 @@ export default class UrlArrayLoader implements IArrayLoader {
     if (!url.endsWith(".json") && !url.endsWith(".parquet")) {
       throw new Error(`Unsupported file format for URL array loader: ${url}`);
     }
-    const { data, min: newMin, max: newMax } = await this.workerPool.exec("load", [url, type]);
-    return new UrlArraySource<T>(data, type, min ?? newMin, max ?? newMax);
+    const { data, texImage, min: newMin, max: newMax } = await this.workerPool.exec("load", [url, type]);
+    // Reconstruct instance of DataTexture on the main thread
+    const tex = typedArrayToDataTexture(texImage.data, type, texImage.width, texImage.height);
```

Comment:
`Texture` has a lot of properties defined on it. If you're just using the data, width, and height here on the main thread side, why not just send those?

Reply:
Did some refactoring, let me know what you think!

Reply:
I like it!